services/classification/luis/client.go

```diff
@@ -0,0 +1,74 @@
+package luis
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/nyaruka/goflow/utils"
+	"github.com/nyaruka/goflow/utils/httpx"
+
+	"github.com/shopspring/decimal"
+)
+
+type ExtractedIntent struct {
+	Intent string          `json:"intent"`
+	Score  decimal.Decimal `json:"score"`
+}
+
+type ExtractedEntity struct {
+	Entity     string          `json:"entity"`
+	Type       string          `json:"type"`
+	StartIndex int             `json:"startIndex"`
+	EndIndex   int             `json:"endIndex"`
+	Score      decimal.Decimal `json:"score"`
+}
+
+type SentimentAnalysis struct {
+	Label string          `json:"label"`
+	Score decimal.Decimal `json:"score"`
+}
+
+// PredictResponse is the response from a predict request
+type PredictResponse struct {
+	Query             string            `json:"query"`
+	TopScoringIntent  *ExtractedIntent  `json:"topScoringIntent"`
+	Intents           []ExtractedIntent `json:"intents" validate:"required"`
+	Entities          []ExtractedEntity `json:"entities"`
+	SentimentAnalysis SentimentAnalysis `json:"sentimentAnalysis"`
+}
```

Comment:
Totally a style call but I prefer to put everything in one struct if I can and there isn't other reuse. Makes it easier to read the entire request as one struct.

Reply:
You mean anonymous structs? Can try that but I've always run into problems then constructing them in tests etc

Reply:
Ya, just a question of usage. If you end up using them independently then makes sense to pull out as you have, but if not having a giant one makes the relationship clearer.